core/io/xml_parser.cpp

```diff
@@ -201,21 +201,58 @@ bool XMLParser::_parse_cdata() {
 
 void XMLParser::_parse_comment() {
 	node_type = NODE_COMMENT;
-	P += 1;
+	ERR_FAIL_COND_MSG(*P != '!', "Bad XML comment");
+	++P;
+	if (*P == '\0') // End of string - exist
+		return;
```

Comment:
You must use braces for all if statements.

Reply:
ok, I will update my fix.